File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) 
    {
        long int t=0,l=0,r=0;
        for(auto i:nums) t=t+i;
        r=t-nums[0];
        l=0;
        for(int j=0;j<nums.size();j++)
        {
            if(l==r) return j;
            else
            {
                l=l+nums[j];
                if(j!=nums.size()-1) r=r-nums[j+1];
            }
        }
        return -1;
        
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        tsum=sum(nums)
        ls=0
        rs=tsum-nums[0]
        for i in range(len(nums)):
            if(ls==rs): return i
            else:
                ls=ls+nums[i]
                if(i!=len(nums)-1):
                    rs=rs-nums[i+1]
                print(ls,rs)
        return -1
        